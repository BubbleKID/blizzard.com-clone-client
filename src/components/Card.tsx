interface Props {
    name: string,
    background: string
}

export default function Card(props: Props) {
    return (
      <img
        src={props.background}
        alt={props.name}
        className="card"
        width="240"
        height="320"
      />
    );
}