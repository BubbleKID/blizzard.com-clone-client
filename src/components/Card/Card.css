.card {
    margin: 12px;
}

.card img {
    width: 210px;
    height: 280px;
    border: 1px solid rgba(0,174,255,0);
    box-shadow: 0 8px 16px rgb(0 0 0 / 10%);
}

.card img:hover {
    border: 1px solid rgba(0,174,255,.5);
    transition-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
    transition-duration: 500ms;  
    filter: brightness(115%) contrast(90%);
    transform: scale(1.05);
}