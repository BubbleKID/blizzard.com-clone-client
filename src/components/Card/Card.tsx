import './Card.css';
import React from 'react';

interface Props {
    name: string,
    background: string
}

export default function Card(props: Props) {
    return (
      <a href="#" className="card">
        <img src={props.background} alt={props.name}/>
      </a>
    );
}