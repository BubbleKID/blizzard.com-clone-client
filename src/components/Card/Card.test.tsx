import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Card from './Card';
import diablo4 from '../../images/game-card_diablo4_enUS.jpeg';

test('Card should be rendered', () => {
    const component = TestRenderer.create(<Card name="card" background={diablo4}/>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});