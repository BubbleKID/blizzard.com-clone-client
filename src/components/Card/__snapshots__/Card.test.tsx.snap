// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be rendered 1`] = `
<a
  className="card"
  href="#"
>
  <img
    alt="card"
    src=""
  />
</a>
`;
